lib/chef/chef_class.rb

```diff
@@ -190,6 +190,22 @@ def provider_handler_map
     def resource_handler_map
       @resource_handler_map ||= Chef::Platform::ResourceHandlerMap.instance
     end
+
+    #
+    # @overload log
+    # Get the current log object.
+    #
+    # @return An object that supports `deprecation(message)`
+    #
+    # @example
+    #     run_context.log.deprecation("Deprecated!")
+    #
+    # @api private
+    def log
```

Comment:
Really uncomfortable calling this `log`, it doesn't return a logging object so the comments are directly wrong.